ibm/resource_ibm_ssl_certificate.go

```diff
@@ -0,0 +1,830 @@
+package ibm
+
+import (
+	fmt "fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	services "github.com/softlayer/softlayer-go/services"
+	session1 "github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const (
+	AdditionalSSLServicesPackageType            = "ADDITIONAL_SERVICES"
+	AdditionalServicesSSLCertificatePackageType = "ADDITIONAL_SERVICES_SSL_CERTIFICATE"
+
+	SSLMask = "id"
+)
+
+func resourceIBMSSLCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMSSLCertificateCreate,
+		Read:     resourceIBMSSLCertificateRead,
+		Update:   resourceIBMSSLCertificateUpdate,
+		Delete:   resourceIBMSSLCertificateDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+
+			"server_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"server_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"validity_months": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"ssl_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"certificate_signing_request": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"renewal_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"order_approver_email_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"technical_contact_same_as_org_address_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"administrative_contact_same_as_technical_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"billing_contact_same_as_technical_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"administrative_address_same_as_organization_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"billing_address_same_as_organization_flag": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"organization_information": {
+				Type:     schema.TypeSet,
+				Required: true,
```

Comment:
Add MaxItems:      1, so it restricts to user to provide organization_information only once similarly for technical_contact, billing_contact, administrative_contact.....

Reply:
Done.